README.md

```diff
@@ -1,222 +1,515 @@
+    // Do something
+  }
+}
 ```
-Or
+
+### Indentation
+
+Two space indentation should be used for all JavaScript and JSON files.
+
+### Manipulating `this` context
+
+Instead of storing context within a variable, functions should be called within the required scope.
+
+Below are examples of code where a property needs to be accessed from a seperate context.
+
+**Don't do this:**
+
 ```javascript
-define(function(require, exports, module)){
-  'strict mode';
-  //code goes here
-});
+class MyClass {
+  constructor() {
+    this.message = 'Hello World!';
+  }
+
+  fetchFoo: function(){
+    // Context is stored in a variable
+    var that = this;
+
+    const getMessage = function(){
+      return that.message;
+    };
+
+    return fetch('some/ajax/request')
+      .then(getMessage);
+  }
+}
 ```
 
-## Unused Variables
+**Do either of these instead:**
+
+```javascript
+class MyClass {
+  constructor() {
+    this.message = 'Hello World!';
+  }
+
+  fetchFoo: function(){
+    return fetch('some/ajax/request')
+      // An arrow function is used to maintain context.
+      .then(() => this.message);
+  }
+}
+```
+
+```javascript
+class MyClass {
+  constructor() {
+    this.message = 'Hello World!';
+  }
 
-Unused variables should be removed from the code when they are no longer needed. This will be [enforced by jshint](http://www.jshint.com/docs/options/#unused).
+  fetchFoo: function(){
+    let getMessage = function(){
+      return this.message;
+    };
 
-Because some libraries expect function arguments with specific names and signatures, function parameters that are not used will not be ignored.
+    // The function is bound to the given context.
+    getMessage = getMessage.bind(this);
+
+    return fetch('some/ajax/request')
+      .then(getMessage);
+  }
+}
+```
+
+```javascript
+class MyClass {
+  constructor() {
+    this.message = 'Hello World!';
+  }
+
+  fetchFoo: function(){
+    // The message is assigned to a local variable
+    const message = this.message
+
+    const getMessage = function(){
+      return message;
+    };
+
+    return fetch('some/ajax/request')
+      .then(getMessage);
+  }
+}
+```
+
+### Curly Braces
+
+Curly braces are required around blocks in loops and conditionals.
 
 **Don't do this:**
+
 ```javascript
-var x;  //x is never used
+if (foo)
+  console.log('bar');
 ```
-**But this is okay:**
+
+**Do this instead:**
+
 ```javascript
-define(function(require, exports, module){
-  //module is never used
-});
+if (foo) {
+  console.log('bar');
+}
 ```
 
-## Coerced Types and Equality
+### Strict Mode
+
+* All modules should be have strict mode enabled.
+* ES6 modules have strict mode enabled by default.
+* Non-ES6 modules should have `'use strict';` placed at the top of the file.
+
+### Coerced Types and Equality
 
-Prefer `===` over `==`. This will be [enforced by JShint](http://www.jshint.com/docs/options/#eqeqeq).
+Strict comparison (`===`) should be used instead of abstract comparison (`==`).
```

Comment:
Exception for checking null? `foo != null` is equivalent to `foo !== null && foo !== undefined`. I think it's a good idiom to adopt and JSHINT has a flag for it.